daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md

```diff
@@ -51,11 +65,64 @@ spec:
   httpPipeline:
     handlers:
     - name: wasm
-      type: middleware.http.wasm.basic
+      type: middleware.http.wasm.httpwasm
+```
+
+### Generating Wasm
+
+This component lets you manipulate an incoming request or serve a response with
+custom logic compiled using the [http-wasm][1] Application Binary Interface
+(ABI). The `handle_request` function receives an incoming request and can
+manipulate it or serve a response as necessary.
+
+To compile your Wasm, you must compile source using a http-handler compliant
+guest SDK such as [TinyGo](https://github.com/http-wasm/http-wasm-guest-tinygo).
+
+Here's an example in TinyGo:
+
+```go
+package main
+
+import (
+	"strings"
+
+	"github.com/http-wasm/http-wasm-guest-tinygo/handler"
+	"github.com/http-wasm/http-wasm-guest-tinygo/handler/api"
+)
+
+func main() {
+	handler.HandleRequestFn = handleRequest
+}
+
+// handleRequest implements a simple HTTP router.
+func handleRequest(req api.Request, resp api.Response) (next bool, reqCtx uint32) {
+	// If the URI starts with /host, trim it and dispatch to the next handler.
+	if uri := req.GetURI(); strings.HasPrefix(uri, "/host") {
```

Comment:
Could you follow invoke API or use substring to check, so that this example can work as expected.

Reply:
This documentation applies to the new, not the old one https://github.com/dapr/components-contrib/pull/2239
I'm required to make a PR on docs before it is merged, that's why this is here now. The example works, just not until the new PR is merged..

Reply:
This documentation applies to the new, not the old one https://github.com/dapr/components-contrib/pull/2239
I'm required to make a PR on docs before it is merged, that's why this is here now. The example works, just not until the new PR is merged..

Reply:
Sure.

Reply:
ok components-contrib is up to date now